cmd/set.go

```diff
@@ -26,10 +27,32 @@ func SetCommand() cli.Command {
 					},
 				},
 				Action: func() {
-					p := fmt.Sprintf("http://localhost:%v", port)
-					a := []string{"config", "set", "proxy", p}
-					e := execCommand{cmd: "npm", args: a}
-					executeCommand(e)
+					p := SetProxyPort(port)
+					cmds := []execute.NewCommand{}
+					cmds = append(cmds, execute.NewCommand{Cmd: "npm", Args: []string{"config", "set", "proxy", p}})
+					execute.Commands(cmds)
+					fmt.Println("Set npm config successfully")
+				},
+			},
+			{
+				Name:        "git",
+				Usage:       "set git proxy config",
+				Description: "additional description?",
+				Flags: []cli.Flag{
+					cli.IntFlag{
+						Name:        "port, p",
+						Value:       3128,
+						Usage:       "set custom CNTLM `PORT`",
+						Destination: &port,
+					},
+				},
+				Action: func() {
+					p := SetProxyPort(port)
+					cmds := []execute.NewCommand{}
+					http := execute.NewCommand{Cmd: "git", Args: []string{"config", "--global", "http.proxy", p}}
```

Comment:
This is all nesting so deeply that this command is scrolling off the side of my screen. A couple of suggestions/questions here:

1) Why is this struct val embedded in a function that returns it? You could just declare the value at the top level and reference that. I know that the way you have it protects the value from being modified - is that really an issue though? Maybe this is what is recommended by the `cli` package you're using; i haven't looked into that yet.

2) You could move the `Action` function out-of-line, and move the `Action:` field above the `Flags` (I'd move it just to keep all the single-line fields together, otherwise `Action` will get visually lost after the flags.

You could also flatten it a bit more by defining the subcommands as vars and referencing them when needed:
```
var (
        set_npm = cli.Command{ ... }
        set_git = cli.Command{ ... }
        set = cli.Command{
            ...
            Subcommands: []cli.Command{set_npm, set_git, ......}
        }
)
```

Also if you turned `execute.NewCommand` into a `[]string` alias, you could more easily construct the command with:
```
var cmd execute.Command = strings.Split("git config --global http.proxy " + p, " ")
```
I usually find it more readable to maintain command lines as a single string if your not doing too much substitution on them. It's less efficient because you're doing the split at run time, but efficiency is not much of a concern here. Could be a line call, so whatever.

Reply:
2. Done, moved above flag declaration.